### PR TITLE
Update Budget Record loading state

### PR DESCRIPTION
**Pull request by frankyma**
Include the update loading in the `ProjectBudget.tsx` loading state

### CODE REVIEW

**Review by CarsonF — COMMENTED** [review]
This just shows the loading skeleton right? I would rather not hide all of the present data just because we are changing one piece.

**Comment by frankyma** [comment]
It's two pieces, the budget line and the total.  When I was testing the update was really quick, do you expect his to be a big enough issue to warrant a specific loading state?

**Comment by CarsonF** [comment]
> It's two pieces, the budget line and the total.

`loading` also goes to the table which I think makes the whole thing loading. Maybe I'm wrong though I haven't checked out the code.

> When I was testing the update was really quick, do you expect his to be a big enough issue to warrant a specific loading state?

Yeah just because the API is fast on your machine doesn't mean that's always going to be the case in every situation.

**Comment by frankyma** [comment]
Changed to only load the row and the total.  I don't know how to add more rows to the table so haven't been able to test that yet.
ezgif com-gif-maker

**Comment by frankyma** [comment]
There's a react warning that was already there 

```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
    in MTableEditCell (created by WithTheme(MTableEditCell))
    in WithTheme(MTableEditCell) (created by MTableBodyRow)
```

I tried to do a little debug bug couldn't solve it.  Not sure if either anyone else have ideas or it's just part of the MUI table

**Comment by glassblowerscat** [comment]
> Not sure if either anyone else have ideas or it's just part of the MUI table

This is a bug in `material-table`; I have an issue open for it

File: src/scenes/Projects/Budget/ProjectBudget.tsx [hunk 1]
```diff
@@ -57,7 +58,12 @@ export const ProjectBudget = () => {
   const { data, loading, error } = useProjectBudgetQuery({
     variables: { id: projectId },
   });
-  const [updateBudgetRecord] = useUpdateProjectBudgetRecordMutation();
+  const [
+    updateBudgetRecord,
+    { loading: isUpdateLoading },
```
Comment (glassblowerscat) [inline src/scenes/Projects/Budget/ProjectBudget.tsx:63]: @frankyma Maybe we should play it safe and use the `onCompleted` method of the mutation to set `budgetRecordIdInUpdate.current` back to `null`; what do you think?

Comment (frankyma) [reply src/scenes/Projects/Budget/ProjectBudget.tsx:63]: Technically it shouldn't be necessary but if you insist on it I don't mind.  Lmk

Comment (glassblowerscat) [reply src/scenes/Projects/Budget/ProjectBudget.tsx:63]: > Technically it shouldn't be necessary but if you insist on it I don't mind. Lmk

Eh, you're right. Just being overly cautious

**Review by glassblowerscat — APPROVED** [review]

**Pull request merged**